src/server/pps/server/api_server.go

```diff
@@ -315,13 +315,10 @@ func (a *apiServer) validateKube() {
 		logrus.Errorf("unable to access kubernetes pods, Pachyderm will continue to work but certain pipeline errors will result in pipelines being stuck indefinitely in \"starting\" state. error: %v", err)
 	}
 	pods, err := a.rcPods("pachd")
-	if err != nil {
+	if err != nil || pods == nil || len(pods) == 0 {
```

Comment:
If `pods == nil` then `len(pods)` is guaranteed to be `0`. So I think you can cut the middle clause.